src/studio/src/designer/DataModeling/Json/Keywords/XsdTotalDigitsKeyword.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using Json.More;
+using Json.Schema;
+using static Altinn.Studio.DataModeling.Utils.RestrictionsHelper;
+
+namespace Altinn.Studio.DataModeling.Json.Keywords;
+
+/// <summary>
+/// Equivalent of totalDigits restriction in json schema
+/// </summary>
+[SchemaKeyword(Name)]
+[SchemaPriority(int.MinValue)]
+[SchemaDraft(Draft.Draft6)]
+[SchemaDraft(Draft.Draft7)]
+[SchemaDraft(Draft.Draft201909)]
+[SchemaDraft(Draft.Draft202012)]
+[JsonConverter(typeof(XsdTotalDigitsKeywordJsonConverter))]
+public sealed class XsdTotalDigitsKeyword: IJsonSchemaKeyword, IEquatable<XsdTotalDigitsKeyword>
+{
+    /// <summary>
+    /// The name of the keyword
+    /// </summary>
+    private const string Name = "totalDigits";
```

Comment:
Note for reviewers:
in json schema this keyword will be represented with "totalDigits" term, not with "@xsd" prefix.
I didn't wanted to tie it closely to xsd since json schema with be build to contain this functionality without context of xsd.